src/test/java/homework_3/ImmutableWorkerTest.java

```diff
@@ -0,0 +1,20 @@
+package homework_3;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class ImmutableWorkerTest {
+
+    @Test
+    void testUpdateImmutable() {
```

Comment:
As it is immutable test, please check immutability, not ability to change object

Reply:
To be honest I'm still not sure how should I test the class' immutability. I refactored the test method but don't know if I did it right.

Reply:
Just create one object, modify it and put to another variable. Test them by ==. Same for link fields for the objects. If they're not same, you should get false everywhere

Reply:
But in general test for ImmutableClass is not needed :)